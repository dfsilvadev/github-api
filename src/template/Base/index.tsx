import { Outlet } from "react-router-dom";

import * as S from "./styles";

const Base = () => {
  return (
    <S.Main aria-label="Main">
      <Outlet />
    </S.Main>
  );
};

export default Base;
