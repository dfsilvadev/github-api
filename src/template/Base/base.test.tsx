import { screen } from "@testing-library/react";
import { renderWithTheme } from "../../utils/tests/renderWithTheme";

import Base from ".";

describe("Base", () => {
  it("should render without error", () => {
    const { container } = renderWithTheme(<Base />);

    expect(screen.getByLabelText("Main")).toHaveStyle({
      width: "100%",
      height: "100vh"
    });

    expect(container).toMatchSnapshot();
  });
});
