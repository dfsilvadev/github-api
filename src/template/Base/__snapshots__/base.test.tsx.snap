// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base should render without error 1`] = `
.c0 {
  width: 100%;
  height: 100vh;
}

<div>
  <main
    aria-label="Main"
    class="c0"
  />
</div>
`;
